ethsigner/core/src/main/java/tech/pegasys/ethsigner/core/requesthandler/sendtransaction/transaction/EeaTransaction.java

```diff
@@ -53,7 +53,7 @@ public void updateNonce(final BigInteger nonce) {
   }
 
   @Override
-  public boolean hasNonce() {
+  public boolean isNonceUserSpecified() {
```

Comment:
What does `isNonceUserSpecified` mean, how is it different / better than `hasNonce`? ...as what is being done is checking whether there is a nonce.

Reply:
Specifically it's checking if the USER gave you the nonce, rather than there being a nonce which may have been inserted via the updateNonce function.

This is required to be known in order to determine if the nonce should be modified prior to transmission